### DISCUSSION TITLE
Add platform support for Radicle

### DISCUSSION THREAD

**Discussion by mzampetakis** [renovatebot/renovate#31803] [Suggest an Idea]
### Tell us more.

[Radicle](https://radicle.xyz/ ) is a sovereign peer-to-peer network for code collaboration, built on top of Git. Radicle enables users to run their own nodes, ensuring censorship-resistant code collaboration and fostering a resilient network without reliance on third-parties. It is under heavy development and currently in its first (v1.0.0) version. Apart from the development of the protocol itself it is pretty crucial for radicle community to integrate it with popular and well-established tooling around it so that it will offer developers an easier adaptation process and a complete solution. Having this integration in place will allow even more developers to use renovate within their radicle repos. Apart from this, developers that have already setup their 
renovate configuration in their repos will be able to transfer them to Radicle and keep their jobs running with minimum effort! With this integration Renovate will also become one of the first dependency management tools to be integrated with Radicle and can position itself as a solution that supports user sovereignty and isn't just compatible with centralized solutions.

I would very much like to work on this integration / implementation and I was wondering if the maintainers will be open in accepting a PR and perhaps providing some help along the way ?

I will implement a new platform for Renovate to support Radicle as the existing ones alongside all the official documentation and tutorials required to support it. As radicle has a different philosophy compared to the already implemented forges there are some limitations. While some issues might arise during the development of this integration there will be significant support form the radicle team to overcome them when feasible. 

I have already seen other implementation for existing forge such as [this one for SCM-manager](https://github.com/renovatebot/renovate/pull/26866) and [this one for Gerrit](https://github.com/renovatebot/renovate/pull/18961/) so, I could rely my work on these already implemented solutions. I have also seen that the "local" platform is not capable of creating branches with the required changes so a new platform has to be implemented

Do you think there is something to take into consideration before starting to implement it. Is there any guide/tutorial (outside the official docs) that could help me on this?

**Comment by rarkins** [comment]
Adding platform is a lot less common/frequent than other modules in Renovate, so we don't really have any guides. The important thing would be making sure that this new platform doesn't require any changes to the way Renovate works which would in turn then be a burden to maintain. Does its concepts map to Renovate's existing ones which are mostly based off the original GitHub platform implementation, e.g. Pull Requests, Issues, Check Runs, etc?

**Reply by mzampetakis** [reply]
> Adding platform is a lot less common/frequent than other modules in Renovate, so we don't really have any guides. 

That makes completely sense!

Radicle supports all of the main functionalities that other forges provide. So, it won't be a burden for its implementation. Having said that, Radicle by its nature has a bit different philosophy as there is not any central point which might be a bit tricky at some cases.

Also, Check Runs is  not supported but as I said it's something we could work on to support it.

Thank you for your feedback!

**Reply by viceice** [reply]
issues and checks are not required for initial support. issues are fully optional (no dashboard / config error issues then).

status checks for commits are required for automerge to work properly.
